### ISSUE TITLE
How to get completion_length?

### ISSUE THREAD

**Issue by Tuziking** [huggingface/trl#3254]
I noticed that during GRPO training, `completion_length` is recorded. However, I found that it’s not simply obtained by `len(completion)`. How is this calculated—by tokens? Is it possible for me to access the `completion_length` for each sample?

**Comment by shirinyamani** [comment]
So if you mean you wanna see the len of each completion generated during training, then the  `completion_length` for each prompt samples is ` completion_mask.sum(-1)`. The raw `completion_length` won't work cuz the completions will further go thro masking process [here](https://github.com/huggingface/trl/blob/1d7b8c4f706a0df9fd4d3e8d0d576e8dd5dedb28/trl/trainer/grpo_trainer.py#L807).

A couple of other notes, 
- you can see prompts and completions with `log_completions=True`
- you can set your desired fixed len for all the completions by setting `max_completion_length` in config (token-based-count), for exp;
```python
training_args = GRPOConfig(....
     max_completion_length=200, )

 trainer = GRPOTrainer(
     model="Qwen/Qwen2.5-0.5B",
     reward_funcs=reward_func,
     args=training_args,
     train_dataset=dataset,
 )
```

**Comment by cuiyuhao1996** [comment]
The way I'm estimating it now is to make sure that Latin is decoupled from the rest of the script.
```
finds = regex.findall(r'[a-zA-Z\p{Script=Latin}\p{Script=Cyrillic}\p{Script=Greek}]+', content)
content = regex.sub(r'[a-zA-Z\p{Script=Latin}\p{Script=Cyrillic}\p{Script=Greek}]+ *', '', content)
completion_len = len(content) + len(finds)
```

**Comment by qgallouedec** [comment]
Hey @Tuziking, sorry for the very late reply, we're catching up on the backlog.

`completion_length` is token-based, not `len(completion)` on the decoded string: it's the number of generated tokens per completion. As @shirinyamani said, per sequence that's the count of model-generated tokens (`completion_mask.sum(-1)` in older versions, or `len(completion_ids)` on current main). It's now logged as `completions/mean_length` / `min_length` / `max_length` (plus terminated-only variants and `completions/clipped_ratio`), see https://github.com/huggingface/trl/blob/44596766101a123f1a3ff70b0aefa54d926179ad/trl/trainer/grpo_trainer.py#L2101-L2103
To inspect the per-sample completions (and their lengths), set `log_completions=True` in `GRPOConfig`, which logs the prompt/completion table to your tracker.
